presto-spi/src/main/java/com/facebook/presto/spi/tracing/TracerProvider.java

```diff
@@ -15,9 +15,12 @@
 
 public interface TracerProvider
 {
+    String getName();
+
+    String getTracerType();
     /**
      *
      * @return A @Tracer that should be kept throughout the whole duration of tracing.
      */
-    Tracer getNewTracer();
+    Tracer getNewTracer(String traceToken, String contextPropagator, String propagatedContext, String baggage);
```

Comment:
[Major] this is my major comment.

Similar to many other "handle" classes in our codebase, let's create a handle interface in `com.facebook.presto.spi.tracing`:
```java
public interface TracerHandle
{
}
```

Change the signature of `getNewTracer` to
```java
Tracer getNewTracer(TracerHandle handle);
```

Create a new method of `TracerProvider`:
```java
/**
 * The method returns a function to take a set of HTTP headers and generate the tracer handle
 */
Function<Set<String>, TracerHandle> getHandleGenerator();
```

In your OpenTelemetry implementation, you should have
```java
public class OpenTelemetryTracerHandle
        implements TracerHandle
{
    private final String traceToken;
    private final String contextPropagator;
    private final String propagatedContext;
    private final String baggage;

    ... 
}
```


In your `OpenTelemetryTracerProvider`, you should have
```java
@Override
public Tracer getNewTracer(TracerHandle handle)
{
    OpenTelemetryTracerHandle tracerHandle = (TracerHandle) handle;
    return new OpenTelemetryTracer(tracerHandle.getXXX, ...);
}
```

In your `getHandleGenerator` implementation, move all the logics you have in presto-main here so we can hide all those headers and specialized extraction methods.